tests/foreman/api/test_convert2rhel.py

```diff
@@ -218,17 +215,39 @@ def oracle(
     act_key = create_activation_key(
         module_target_sat, module_entitlement_manifest_org, module_lce, cv, c2r_sub.id
     )
+    # UBI repo required for subscription-manager packages on Oracle
     ubi_url = settings.repos.convert2rhel.ubi7 if major == '7' else settings.repos.convert2rhel.ubi8
-    ubi = create_repo(module_target_sat, module_entitlement_manifest_org, ubi_url)
-    ubi_repo = ubi.full_path.replace('https', 'http')
-    register_host(
-        module_target_sat,
-        act_key,
-        module_entitlement_manifest_org,
-        smart_proxy_location,
-        oracle_host,
-        ubi_repo,
-    )
+
+    # Register Oracle host with Satellite
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_entitlement_manifest_org,
+        activation_keys=[act_key.name],
+        location=smart_proxy_location,
+        insecure=True,
+        repo=ubi_url,
+    ).create()
+    assert oracle_host.execute(command).status == 0
+
+    if major == '8':
+        # needs-restarting missing in OEL8
```

Comment:
possibly make this future-proof and assume this will be the case in rel. `9` as well.
```suggestion
    if int(major) >= 8:
        # needs-restarting missing since OEL8
```

Reply:
Same here. Also, some remediations for EL9 could be same or could change, so I think we can consider fixing that later, Thank you!